libs/agno/agno/skills/agent_skills.py

```diff
@@ -0,0 +1,216 @@
+        self.loaders = loaders
+        self._skills: Dict[str, Skill] = {}
+        self._loaded = False
+
+    def _ensure_loaded(self) -> None:
+        """Ensure skills are loaded from all loaders."""
+        if self._loaded:
+            return
+
+        for loader in self.loaders:
+            try:
+                skills = loader.load()
+                for skill in skills:
+                    if skill.name in self._skills:
+                        log_warning(f"Duplicate skill name '{skill.name}', overwriting with newer version")
+                    self._skills[skill.name] = skill
+            except Exception as e:
+                log_warning(f"Error loading skills from {loader}: {e}")
+
+        self._loaded = True
+        log_debug(f"Loaded {len(self._skills)} total skills")
+
+    def get_skill(self, name: str) -> Optional[Skill]:
+        """Get a skill by name.
+
+        Args:
+            name: The name of the skill to retrieve.
+
+        Returns:
+            The Skill object if found, None otherwise.
+        """
+        self._ensure_loaded()
+        return self._skills.get(name)
+
+    def get_all_skills(self) -> List[Skill]:
+        """Get all loaded skills.
+
+        Returns:
+            A list of all loaded Skill objects.
+        """
+        self._ensure_loaded()
+        return list(self._skills.values())
+
+    def get_skill_names(self) -> List[str]:
+        """Get the names of all loaded skills.
+
+        Returns:
+            A list of skill names.
+        """
+        self._ensure_loaded()
+        return list(self._skills.keys())
+
+    def get_system_prompt_snippet(self) -> str:
+        """Generate a system prompt snippet with available skills metadata.
+
+        This creates an XML-formatted snippet that provides the agent with
+        information about available skills without including the full instructions.
+
+        Returns:
+            An XML-formatted string with skills metadata.
+        """
+        self._ensure_loaded()
+
+        if not self._skills:
+            return ""
+
+        lines = ["<available_skills>"]
+        for skill in self._skills.values():
+            lines.append("<skill>")
+            lines.append(f"  <name>{skill.name}</name>")
+            lines.append(f"  <description>{skill.description}</description>")
+            if skill.scripts:
+                lines.append(f"  <scripts>{', '.join(skill.scripts)}</scripts>")
+            if skill.references:
+                lines.append(f"  <references>{', '.join(skill.references)}</references>")
+            lines.append("</skill>")
+        lines.append("</available_skills>")
+        lines.append("")
+        lines.append(
+            "You have access to skills that provide domain expertise. Use the get_skill_instructions tool "
+            "to load full instructions when you need to use a skill. Use get_skill_reference to access "
+            "detailed documentation from reference files."
+        )
```

Comment:
Let's put this first. We may need to explain a bit further what skills are and the fact skill info should be discoverable gradually. Maybe there are some conventinoal prompts around already